cmd/repo_init.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"os"
+	"time"
+
+	"github.com/pingcap-incubator/tiup/pkg/localdata"
+	"github.com/pingcap-incubator/tiup/pkg/meta"
+	"github.com/pingcap-incubator/tiup/pkg/repository"
+	"github.com/pingcap-incubator/tiup/pkg/utils"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func newRepoInitCmd(env *meta.Environment) *cobra.Command {
+	var (
+		pubKey  string // public key of root
+		privKey string // private key of root
+	)
+	cmd := &cobra.Command{
+		Use:   "init [path]",
+		Short: "Initialise an empty repository",
+		Long: `Initialise an empty TiUP repository at given path. If path is not specified, the
+current working directory (".") will be used.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 1 {
+				repoPath = args[0]
+			}
+
+			// create the target path if not exist
+			if utils.IsNotExist(repoPath) {
+				var err error
+				if err = os.Mkdir(repoPath, 0755); err != nil {
+					return err
+				}
+			}
+			// init requires an empty path to use
+			empty, err := utils.IsEmptyDir(repoPath)
+			if err != nil {
+				return err
+			}
+			if !empty {
+				return errors.Errorf("the target path '%s' is not an empty directory", repoPath)
+			}
+
+			return initRepo(repoPath)
+		},
+	}
+
+	cmd.Flags().StringVar(&pubKey, "pubkey", "", "Path to the public key file")
+	cmd.Flags().StringVar(&privKey, "privkey", "", "Path to the private key file")
+
+	return cmd
+}
+
+func initRepo(path string) error {
```

Comment:
I think it is good for the functions in `cmd` to do the minimum and mostly just call out to higher-level functions, in this case in manifest.go, that makes it easier to change the interface or add alternate interfaces and probably easier to test too.

With the changes I suggested, this function might end up looking much smaller, but if not, could you try and move more to manifest.go please?